viskores/cont/CubicHermiteSpline.h

```diff
@@ -0,0 +1,97 @@
+//============================================================================
+//  The contents of this file are covered by the Viskores license. See
+//  LICENSE.txt for details.
+//
+//  By contributing to this file, all contributors agree to the Developer
+//  Certificate of Origin Version 1.1 (DCO 1.1) as stated in DCO.txt.
+//============================================================================
+
+//============================================================================
+//  Copyright (c) Kitware, Inc.
+//  All rights reserved.
+//  See LICENSE.txt for details.
+//
+//  This software is distributed WITHOUT ANY WARRANTY; without even
+//  the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+//  PURPOSE.  See the above copyright notice for more information.
+//============================================================================
+#ifndef viskores_cont_CubicHermiteSpline_h
+#define viskores_cont_CubicHermiteSpline_h
+
+#include <viskores/cont/viskores_cont_export.h>
+#include <viskores/Range.h>
+#include <viskores/Types.h>
+#include <viskores/cont/ArrayHandle.h>
+#include <viskores/cont/ExecutionObjectBase.h>
+#include <viskores/cont/UnknownArrayHandle.h>
+#include <viskores/exec/CubicHermiteSpline.h>
+
+namespace viskores
+{
+namespace cont
+{
+
+class VISKORES_CONT_EXPORT CubicHermiteSpline : public viskores::cont::ExecutionObjectBase
+{
+public:
+  CubicHermiteSpline() = default;
+  virtual ~CubicHermiteSpline() = default;
+
+  VISKORES_CONT void SetData(const viskores::cont::ArrayHandle<viskores::Vec3f>& data) { this->Data = data; }
+  VISKORES_CONT void SetData(const std::vector<viskores::Vec3f>& data,
+                         viskores::CopyFlag copy = viskores::CopyFlag::On)
+  {
+    this->Data = viskores::cont::make_ArrayHandle(data, copy);
```

Comment:
When setting the data, shouldn't the knots and tanget arrays be cleared out? That way you will be sure to recompile them if necessary.